skyportal/handlers/public/source_page.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+from sqlalchemy.orm import scoped_session, sessionmaker
+import sqlalchemy as sa
+
+from baselayer.app.env import load_env
+from baselayer.app.models import DBSession
+from ...models.public_pages.public_source_page import PublicSourcePage
+
+from ...utils.cache import Cache
+from ..base import BaseHandler
+
+env, cfg = load_env()
+
+Session = scoped_session(sessionmaker())
+
+cache_dir = "cache/public_pages/sources"
+cache = Cache(
+    cache_dir=cache_dir,
+    max_age=cfg["misc.minutes_to_keep_public_source_pages_cache"] * 60,
+)
+
+
+def get_version(source_id, version_date):
+    if Session.registry.has():
+        session = Session()
+    else:
+        session = Session(bind=DBSession.session_factory.kw["bind"])
+    query = sa.select(PublicSourcePage).where(
+        PublicSourcePage.source_id == source_id, PublicSourcePage.is_public
+    )
+    if version_date:
+        query = query.where(PublicSourcePage.created_at_iso == version_date)
+    else:
+        query = query.order_by(PublicSourcePage.created_at.desc())
+    return session.scalars(query).first()
+
+
+class SourcePageHandler(BaseHandler):
+    def get(self, source_id=None, version_date=None):
+        """
+        ---
+        description: Display the public page for a given source and version
```

Comment:
Usually for API handlers that return multiple things as a list, or just one (if ID is provided), we have a specific dosctring syntax with the "single" and "multiple" keywords. Please correct that here before merging.

Reply:
✅ Done.